### PR TITLE
make the website functional and user-friendly for now until it's time for a full rewrite!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

### PR DESCRIPTION
**Pull request by joeturki**
this PR doesn't address all the CSS or UX issues, but it makes the website usable on mobile and smaller monitors. 
it also fixes layout problems and random issues like unnecessary double scrollbars.
i think I've tested all the pages, and the UX is ok for now in most pages. most likely gonna submit another PR before the full rewrite.

### CODE REVIEW

File: src/components/twitchChat.vue [hunk 1]
```diff
@@ -1,26 +1,23 @@
...
-          style="border-radius: 15px; overflow: hidden;">
-        </iframe>
-    </div>
+	<div class="twitch-iframe">
+		<h2>Ask for help, or try it out!</h2>
+
+		<iframe id="twitch-chat-embed"
+    	:src="`https://www.twitch.tv/embed/potatbotat/chat?parent=${host}&darkpopout`"
+			height="550"
+			width="100%"
```
Comment (RyanPotat) [inline src/components/twitchChat.vue:14]: pls add back 'style="border-radius: 15px;">' inline or in the twitch-chat-embed id

Comment (JoTurkk) [reply src/components/twitchChat.vue:14]: ok it's back, it's deadass ugly tho (atleast for me).

**Review by RyanPotat — COMMENTED** [review]
I like the changes especially the routing, however I did like giving users the top level visibility of categories to quickly find commands.

As it is here for a user to 'discover' a command in, lets say the anime section, they'd have to scroll nearly all the way down 100+ commands. A dropdown category selector or similar would be nice to make it more navigable.

I meant to comment this on the `src/views/help.vue` file :erm:

**Comment by RyanPotat** [comment]
Is there a way to stop the sidebar from popping in and out when clicking on an element that overflows? Or at least stop it from shifting the background image?

**Comment by JoTurkk** [comment]
i suspect that the vast majority of users wanna just read about specific command or few commands  Ctrl+F or search. normal users are not going to read through 10+ commands in one category.

and for the users that wanna discover the bot this is also better than switching between categories. you could run an A/B test, if you don't believe me. Double menus are a turn-off, but I can make it so that categories can be collapsed

**Comment by JoTurkk** [comment]
> Is there a way to stop the sidebar from popping in and out when clicking on an element that overflows? Or at least stop it from shifting the background image?

which sidebar?

**Comment by RyanPotat** [comment]
> i suspect that the vast majority of users wanna just read about specific command or few commands Ctrl+F or search. normal users are not going to read through 10+ commands in one category.
> 
> and for the users that wanna discover the bot this is also better than switching between categories. you could run an A/B test, if you don't believe me. Double menus are a turn-off, but I can make it so that categories can be collapsed

Collapsible categories would be perfect

**Comment by RyanPotat** [comment]
It looks like there is still a double scroll bar on the route `/emotes/history/:channelName`

**Comment by RyanPotat** [comment]
> > Is there a way to stop the sidebar from popping in and out when clicking on an element that overflows? Or at least stop it from shifting the background image?
> 
> which sidebar?

In the help view, if you click on any command which has a long enough description to overflow vertically, the scrollbar appearing causes all elements on the page to shift the scrollbar's width, including the background image

**Comment by JoTurkk** [comment]
> It looks like there is still a double scroll bar on the route `/emotes/history/:channelName`

i didn't fix that one

**Comment by JoTurkk** [comment]
> > > Is there a way to stop the sidebar from popping in and out when clicking on an element that overflows? Or at least stop it from shifting the background image?
> > 
> > 
> > which sidebar?
> 
> In the help view, if you click on any command which has a long enough description to overflow vertically, the scrollbar appearing causes all elements on the page to shift the scrollbar's width, including the background image

can you send a screenshot

**Comment by RyanPotat** [comment]
> > > > Is there a way to stop the sidebar from popping in and out when clicking on an element that overflows? Or at least stop it from shifting the background image?
> > > 
> > > 
> > > which sidebar?
> > 
> > 
> > In the help view, if you click on any command which has a long enough description to overflow vertically, the scrollbar appearing causes all elements on the page to shift the scrollbar's width, including the background image
> 
> can you send a screenshot

image
image
When the scrollbar spawns in it shifts over all elements to the left.

**Comment by JoTurkk** [comment]
oh yeah, yeah this is sadly not fixable without ugly hacks, but it's normal, there was overflow: overlay but it got dropped https://caniuse.com/css-overflow-overlay

File: src/components/twitchChat.vue (hunk 1 above)
Comment (RyanPotat) [reply src/components/twitchChat.vue:14]: also text-align: center; for the twitch-iframe pls

**Comment by RyanPotat** [comment]
Not to be too nit-picky, but I kind of liked this layout for the home page components, and the fixed component size + larger text size of the join/part button. 

Also I think simply 'Join' and 'Part' are okay too, the green and red alone are pretty self explanatory.
image

**Comment by JoTurkk** [comment]
the problem with this layout is that it only works on very large viewports

**Comment by JoTurkk** [comment]
also twitch chat  is more natural  on the right+ it's true that the join button is fine but if you do change the layout i think join my channel is better

**Comment by JoTurkk** [comment]
tbh the homepage looks ugly to me and it needs a rework, the new is worse but it's just more usable, if you wanna stick with the old one let me know and i will try keep it (except for the  "the fixed component size" part erm)

**Comment by RyanPotat** [comment]
> also twitch chat is more natural on the right+ it's true that the join button is fine but if you do change the layout i think join my channel is better


I don't disagree. We can leave that as is.

The only other thing I see would be centering the text above the twitch chat embed

**Comment by JoTurkk** [comment]
feel free to change any text, also you can play with the grid (#home-container)  if you wanna see other options for the layout

**Comment by JoTurkk** [comment]
btw i committed [Collapsible menus](https://github.com/RyanPotat/potat-site-v2/pull/3/commits/c9c0d0e2289bf52415194faddaa990369732d4e7) if you didn't see

**Comment by RyanPotat** [comment]
ZraYP

**Pull request merged**